Peng_Robinson_Class.m

```diff
@@ -0,0 +1,280 @@
+                %Defining the reduced form:
+                
+                A_card = 1;
+                B_card = -(1 - B);
+                C_card = +(A - 3*B.^2 - 2*B);
+                D_card = -(A.*B - B.^2 - B.^3);
+                
+                a_card = B_card ./ A_card;
+                b_card = C_card ./ A_card;
+                c_card = D_card ./ A_card;
+                p_card = b_card - (a_card.^2 / 3);
+                q_card = (2*a_card.^3 / 27) - (a_card.*b_card/3) + c_card;
+                
+                %Solving the Reduced Form
+                
+                delta_card = (q_card/2).^2 + (p_card/3).^3;
+                u = (-q_card/2 + delta_card.^0.5).^(1/3);
+                v = (-q_card/2 - delta_card.^0.5).^(1/3);
+                
+                epsilon_1 = -0.5 + 0.5i * 3^0.5;
+                epsilon_2 = epsilon_1^2;
+                
+                %Differenting betweeen the delta_card values. Namley < or
+                %< than 0. This is done with logical indexing in the
+                %following few lines
+                
+                Z = nan(size(P,1),3);
+                L_Case1 = delta_card <= 0;          %Case if p_card < 0 and delta_card <= 0:  3 real solutions
+                L_Case2 = delta_card > 0;           %CAse if p_card < 0 and delta_card > 0 :  1 real and 2 complex solutions
+                
+                
+                Z(L_Case2,1) = (u(L_Case2)+v(L_Case2))-a_card(L_Case2)/3;
+                Z(L_Case2,2) = (u(L_Case2)*epsilon_1 + v(L_Case2)*epsilon_2)-a_card(L_Case2)/3;
+                Z(L_Case2,3) = (u(L_Case2)*epsilon_2 + v(L_Case2)*epsilon_1)-a_card(L_Case2)/3;
+                
+                Z(L_Case1,1) = ( sqrt(-(4/3).*p_card(L_Case1)).*cos((1/3).*acos(-(q_card(L_Case1)./2).*sqrt(-27./p_card(L_Case1).^3))))       - a_card(L_Case1)./3;
+                Z(L_Case1,2) = (-sqrt(-(4/3).*p_card(L_Case1)).*cos((1/3).*acos(-(q_card(L_Case1)./2).*sqrt(-27./p_card(L_Case1).^3))+pi./3)) - a_card(L_Case1)./3;
+                Z(L_Case1,3) = (-sqrt(-(4/3).*p_card(L_Case1)).*cos((1/3).*acos(-(q_card(L_Case1)./2).*sqrt(-27./p_card(L_Case1).^3))-pi./3)) - a_card(L_Case1)./3;
+                
+                Vm = (T.*Z.*R)./P;
+                vm_liquid = min(Vm,[],2);
+                vm_vapor  = max(Vm,[],2);
+                Vm = [vm_liquid, vm_vapor];
+                
+                % Calculation of fugacities
+                phi = @(Z) exp(Z - 1 - log(Z - B) - A./(2*2^0.5*B).*log((Z+2.414*B)./(Z-0.414*B)));
+                
+                Z_liquid = P.*vm_liquid ./ (R*T);
+                Z_vapor = P.*vm_vapor ./ (R*T);
+                phi_liquid = phi(Z_liquid);
+                phi_vapor = phi(Z_vapor);
+                
+                converged = (abs(phi_liquid - phi_vapor)) <= 0.01;         %!!!!!!!!!!!!!!!!!!!!!!!!
+                
+                T = T.*phi_liquid./phi_vapor;                              %!!!!!!!!!!!!!!!!!!!!!!!!
+                

```

Comment:
These are the essential lines where this function fails. Because the new calculated T is not converging, the following  iterations only give back very bad numbers. Even NaN after some iterations